src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,32 @@
+import java.util.Arrays;
+
+public class ArrayValueCalculator {
+    public static void main(String[] args) {
+
+        try {
+            int result = doCalc(new String[][]{{"2", "4", "5", "6"}, {"5", "6", "7", "7"}, {"9", "10", "11", "8"}, {"9", "5", "1", "14"}});
+            System.out.println("Summ = " +result);
+        } catch (ArraySizeException | ArrayDataException exception) {
+            System.out.println(exception.getMessage());
+        }
+
+    }
+
+    static int doCalc(String [][] arrayNumbers) {
+        int sum = 0;
+        for (int i = 0; i <= arrayNumbers.length -1; i++ ) {
+            for (int j = 0; j <= arrayNumbers[i].length -1; j++) {
+                if(arrayNumbers.length !=4 || arrayNumbers[i].length !=4) throw new ArraySizeException();
```

Comment:
In your solution possible to throw ArrayDataException for an invalid array size (ArraySizeException should be thrown because the array size is invalid, but we check before cells format). Example
String[][] arr = {
{"1", "XXX", "567", "1234"},
{"45"},
{"45", "998", "567", "1234"},
{"45", "998", "567", "456"}
};